### PR TITLE
monitoring: add telemetrygatewayexporter panels, improve metrics

### PR DESCRIPTION
**Pull request by bobheadxi**
Part of https://github.com/sourcegraph/sourcegraph/issues/56970 - this adds some dashboards for the export side of things, as well as improves the existing metrics. Only includes warnings.

## Test plan

Had to test locally only because I ended up changing the metrics a bit, but validated that the queue size metric works in S2.

Testing locally:

```yaml
# sg.config.overwrite.yaml
env:
  TELEMETRY_GATEWAY_EXPORTER_EXPORT_INTERVAL: "30s"
  TELEMETRY_GATEWAY_EXPORTER_EXPORTED_EVENTS_RETENTION: "5m"
  TELEMETRY_GATEWAY_EXPORTER_QUEUE_CLEANUP_INTERVAL: "10m"
```

```
sg start
sg start monitoring
```

Do lots of searches to generate events. Note `telemetry-export` feature flag must be enabled

Data is not realistic because of the super high interval I configured for testing, but it shows that things work:

image

### CODE REVIEW

File: enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go [hunk 1]
```diff
@@ -38,14 +38,21 @@ func (c *config) Load() {
 
 	c.ExportInterval = env.MustGetDuration("TELEMETRY_GATEWAY_EXPORTER_EXPORT_INTERVAL", 10*time.Minute,
 		"Interval at which to export telemetry")
+	if c.ExportInterval > 1*time.Hour {
```
Comment (unknwon) [inline enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go:41]:
```suggestion
	if c.ExportInterval > time.Hour {
```

nit? 🤔

File: enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go [hunk 2]
```diff
@@ -38,14 +38,21 @@ func (c *config) Load() {
...
+		c.AddError(errors.New("TELEMETRY_GATEWAY_EXPORTER_EXPORT_INTERVAL cannot be more than 1 hour"))
+	}
+
 	c.MaxExportBatchSize = env.MustGetInt("TELEMETRY_GATEWAY_EXPORTER_EXPORT_BATCH_SIZE", 5000,
 		"Maximum number of events to export in each batch")
+	if c.MaxExportBatchSize < 100 {
+		c.AddError(errors.New("TELEMETRY_GATEWAY_EXPORTER_EXPORT_BATCH_SIZE must be greater than 100"))
```
Comment (unknwon) [inline enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go:48]: "greater than" or "no less than"?

**Review by unknwon — APPROVED** [review]
Otherwise great dashboard 😁

File: enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go (hunk 2 above)
Comment (bobheadxi) [reply enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go:48]:
```suggestion
	if c.MaxExportBatchSize < 100 {
		c.AddError(errors.New("TELEMETRY_GATEWAY_EXPORTER_EXPORT_BATCH_SIZE must be no less than 100"))
```

File: enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go (hunk 1 above)
Comment (bobheadxi) [reply enterprise/cmd/worker/internal/telemetrygatewayexporter/telemetrygatewayexporter.go:41]: I think I might leave this one as-is, my brain needs the number to read `one hour` and not just `hour` 😆

**Pull request merged**